Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -354,16 +354,34 @@ public void RegisterControlLayout(string json, string name = null, bool isOverri
             }
 
             // Add it to our records.
-            var isReplacement = DoesLayoutExist(internedLayoutName);
+            var isReplacement = m_Layouts.HasLayout(internedLayoutName); 
+            if (isReplacement && isOverride)
+            {   // Do not allow a layout override to replace a "base layout" by name, but allow layout overrides
+                // to replace an existing layout override.
+                // This is required to guarantee that its a hierarchy (directed graph) rather
+                // than a cyclic graph.
+                
+                var isReplacingOverride = m_Layouts.layoutOverrideNames.Contains(internedLayoutName);
+                if (!isReplacingOverride)
+                {
+                    throw new ArgumentException($"Failed to register layout override '{internedLayoutName}'" +
+                                                $"since a layout named '{internedLayoutName}' already exist. Layout overrides must " +
+                                                $"have unique names with respect to existing layouts.");    
+                }
+            }
+
             m_Layouts.layoutStrings[internedLayoutName] = json;
             if (isOverride)
             {
-                m_Layouts.layoutOverrideNames.Add(internedLayoutName);
+                m_Layouts.layoutOverrideNames.Add(internedLayoutName); // set semantics
```

Comment:
What does "set semantics" mean?

Reply:
I should maybe get rid of it, the code is valid since the underlying container is a set (dictionary) if it was e.g. a list it would be a bug

Reply:
removed it